### PR TITLE
Update the Orb development kit

### PR DESCRIPTION
**Pull request by cortinico**
Adapted from https://github.com/CircleCI-Public/Orb-Template

Skipped a few best practices:

* RC002 – All components (jobs, commands, executors, examples) must have descriptions (skipped)
* RC005 – Write a detailed orb description. (skipped)
* RC007 – Home URL should be valid. (skipped)
* RC009 – Complex Run step's commands should be imported. (skipped)

Note: RC007 is skipped because there is no `home_url` present in `@orb.yml` ([source](https://github.com/CircleCI-Public/orb-tools-orb/blob/v12.0.0/src/scripts/review.bats#L91))

### CODE REVIEW

**Comment by fotos** [comment]
@cortinico 

What's pending in this PR:

1. A way to trigger a release. Previously merging into `master` was triggering a new release, using `semantic-release`. This doesn't exist in the current default flow from the template. Need a bit more effort to get this done.

What can be improved in the PR:

1. I fixed a few linting issues but also skipped a few more (missing descriptions, improving description and extracting scripts). I believe all that can be done in a subsequent PR.
2. Testing more commands – right now there is a "test" for `yarn_install` only.

**Comment by cortinico** [comment]
> 1. A way to trigger a release. Previously merging into `master` was triggering a new release, using `semantic-release`. This doesn't exist in the current default flow from the template. Need a bit more effort to get this done.

According to documentation releases are triggered whenever we publish a new tag/release no?
https://circleci.com/docs/creating-orbs/

I don't think we need any token or so setup

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,72 +1,28 @@
...
-          requires: [pack]
-          filters:
-            branches:
-              only: master
+      - orb-tools/lint:
+          filters: *filters
+      - orb-tools/pack:
+          filters: *filters
+      - orb-tools/review:
+          exclude: RC002,RC005,RC009
```
[the remark this answers is not in the document]
Comment (fotos) [reply .circleci/config.yml:19]: The link is from change into `main` changes, which are [not released yet](https://github.com/CircleCI-Public/Orb-Template/commit/eb71a3b1330ac94ee17f81bd2cb92dbb33a8fa98).

This PR is using [`orb-tools/review@12.0`](https://github.com/CircleCI-Public/Orb-Template/blob/v2.0.2/.circleci/config.yml) which doesn't require the `orb_name` parameter.

**Comment by fotos** [comment]
> > 1. A way to trigger a release. Previously merging into `master` was triggering a new release, using `semantic-release`. This doesn't exist in the current default flow from the template. Need a bit more effort to get this done.
> 
> According to documentation releases are triggered whenever we publish a new tag/release no? https://circleci.com/docs/creating-orbs/
> 
> I don't think we need any token or so setup

I also believe the tokens, which currently exist in the CircleCI environment, can be removed.

What I'm taking about above is something different. If you [look at the documentation closely](https://circleci.com/docs/creating-orbs/) it says:

> To publish a new version of our orb, you need to tag your release. A tag can be created and pushed manually, however we recommend using [GitHub.com’s Releases feature](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release).

Basically the orb publishing guide suggests triggering a [release "manually" through the GH UI](https://github.com/react-native-community/react-native-circleci-orb/releases/new). Currently the project supports `semantic-release` which also [updates the changelog and creates the GH release](https://github.com/react-native-community/react-native-circleci-orb/blob/v7.3.0/.releaserc). I think we should maintain that.

All in all the flow should be:

* Merging into `main` triggers a CI/CD release which uses `semantic-release` to create a new tag
* The new tag is automatically [triggering the `orb-tools/publish` workflow](https://github.com/react-native-community/react-native-circleci-orb/pull/177/files#diff-1d3c0150af90ce1dd4e8ff2ce93160721d21a31790a051a5ab2b6aac49704535R35-R43) based on the [`filters`](https://github.com/react-native-community/react-native-circleci-orb/pull/177/files#diff-1d3c0150af90ce1dd4e8ff2ce93160721d21a31790a051a5ab2b6aac49704535R11-R16).

That way merging into `main` will automatically trigger a release which will update the tag, the changelog, the GH release, and finally trigger the publish workflow. I believe that's the way forward. I'll get this done later today.

**Comment by cortinico** [comment]
> That way merging into `main` will automatically trigger a release which will update the tag, the changelog, the GH release, and finally trigger the publish workflow. I believe that's the way forward. I'll get this done later today.

Yup that sounds reasonable to me 👍

**Comment by cortinico** [comment]
Are we fine merging this after [ec0edd4](https://github.com/react-native-community/react-native-circleci-orb/pull/177/commits/ec0edd46c6e085aba7c46cb6a463a82c6b82ff92) ?

**Comment by fotos** [comment]
> Are we fine merging this after [ec0edd4](https://github.com/react-native-community/react-native-circleci-orb/pull/177/commits/ec0edd46c6e085aba7c46cb6a463a82c6b82ff92) ?

Heh, I was just writing that.

I think [that](https://github.com/react-native-community/react-native-circleci-orb/pull/177/commits/ec0edd46c6e085aba7c46cb6a463a82c6b82ff92) should do it.

I'll go ahead and delete the ENV variables defined in CircleCI.

I believe this is good to go. It's hard to test the whole release flow anyway so let's get this merged and see what happens. 🤞

**Review by fotos — APPROVED** [review]
👍 LGTM 🚀

**Comment by fotos** [comment]
If something doesn't go as expected – I'll do a quick follow up PR.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,72 +1,39 @@
 version: 2.1
-
+setup: true
```
Comment (cortinico) [inline .circleci/config.yml:2]: This is making the CI fail on `master`. Could you look into it @fotos ?

Comment (fotos) [reply .circleci/config.yml:2]: Will do tomorrow (CET).

Comment (fotos) [reply .circleci/config.yml:2]: Indeed having `setup: true` allows [only one workflow to defined](https://support.circleci.com/hc/en-us/articles/360060934851-Troubleshooting-Error-Message-Max-number-of-workflows-exceeded).

Funnily enough I did test the `configuration` before I said it's ready to merge:

```sh
$ circleci config validate
Config file at .circleci/config.yml is valid.
```

OK, I'll refactor this and open a new PR.

Comment (fotos) [reply .circleci/config.yml:2]: Follow up PR is #182.

**Pull request merged**